monitoring_tools.py

```diff
@@ -0,0 +1,231 @@
+       args:
+            txt_file: path to txt file to read
+            line_num: the line number after which to insert a new line/lines of text
+            string: the actual line of text you wish to insert."""
+    with open(txt_file, "r") as file:
+        lines = file.readlines()
+
+    lines.insert(line_num, string + "\n")
+
+    with open(txt_file, "w") as file:
+        file.writelines(lines)
+
+
+def create_summary_term_list(value_list, config):
+    if value_list is None:
+        return None
+    else:
+        terms = ""
+        for index, term in enumerate(config.SUMMARY_TERMS):
+            terms += f"<li>{term}: {value_list[index]}</li>"
+        return terms
+
+def add_format_batch_report(num_records, uploader, md5_code, custom_terms=None):
+    """add_format_batch_report:
+        creates template of upload batch report. Takes standard summary terms and args,
+       and allows for custom terms to be added with custom_terms.
+       args:
+            num_records: the number of records uploaded to DB
+            uploader: agent id , or name of uploader.
+            md5_code: the md5 code of this current upload batch.
+            config: the config file to use.
+            custom_terms: the list of custom values to add as summary terms, myst correspond with order of
+                          SUMMARY_TERMS variable in config."""
+
+    if custom_terms is None:
+        custom_terms = ""
+    else:
+        pass
+
+    report = """<html>
+    <head>
+        <title>Upload Batch Report</title>
+        <style>
+        img {
+            max-width: 300px; /* Maximum width of 300 pixels */
+            max-height: 300px; /* Maximum height of 200 pixels */
+        }
+                table {
+            table-layout: fixed;
+            border-collapse: collapse;
+            width: 100%;
+        }
+
+        table, th, td {
+            border: 1px solid black;
+        }
+
+        th, td {
+            padding: 8px;
+            white-space: nowrap;
+            overflow: hidden;
+            text-align: left;
+        }
+    </style>
+    </head>
+    """ + f"""<body>
+        <h1>Upload Batch Report</h1>
+        <hr>
+        <p>Date and Time: {time_utils.get_pst_time_now_string()}</p>
+        <p>Uploader: {uploader}</p>
+        <p>Batch md5: {md5_code}</p>
+
+        <h2>Summary Statistics:</h2>
+        <ul>
+            <li>Number of Records Added: {num_records}</li>
+            {custom_terms}
+        </ul>
+        <h2>Summary Figures:</h2>
+        <h2>Images Uploaded:</h2>
+        <table>
+            <tr>
+                <th style="width: 50%">File Path</th>
+                <th>Barcode</th>
+                <th>Success</th>
+            </tr>
+            
+        </table>
+        
+    </body>
+    </html>
+    """
+
+    add_line_between("import_monitoring.html", line_num=0, string=report)
+
+
+def create_monitoring_report(num_barcodes, batch_md5, agent, config_file, value_list=None):
```

Comment:
for now num_barcodes processed is used instead of image batch size, since it was easier to calculate in-line. For picturae i am aware that each image will have a new row, but I'd have to see how other collections organize their barcodes with multiple images.

Reply:
do it with quesries instead of running tallies

Reply:
limit to the data entryu operator specified (currently there's an example in your template)